src/pip/_internal/req/req_install.py

```diff
@@ -431,7 +431,9 @@ def setup_py(self):
 
         # Python2 __file__ should not be unicode
         if six.PY2 and isinstance(setup_py, six.text_type):
-            setup_py = setup_py.encode(sys.getfilesystemencoding())
+            fs_enc = sys.getfilesystemencoding()
+            if fs_enc is not None:
+                setup_py = setup_py.encode(fs_enc)
```

Comment:
This is wrong - whether `setup_py` is bytes or Unicode depends on whether `fs_enc` is `None`. Even if this doesn't cause tests to fail, it will cause maintenance issues in the long run.